src/resolvers/index.ts

```diff
@@ -22,10 +23,27 @@ const noPerms = {
 
 export const resolvers: Resolvers<Context> = {
   Query: {
-    me: (_, _params, ctx) => {
+    me: async (_, _params, ctx) => {
+      let username;
       if (!ctx.session.user) return null;
-      console.log(ctx.session.user.username);
-      return null;
+      else if (!ctx.session.user.username) return null;
```

Comment:
These could be merged using `&&` I think. I didn't know that the `username` property was optional.

Reply:
&& doesn't work as one is dependant on the other

Reply:
Is the type checker telling you this? It's pretty smart about checking this iirc?

Reply:
Yes I think, when i do if (!ctx.session.user && **!ctx.session.user**.username ) return null; The bolded part is the error as it says "Object is possibly 'undefined'"

Reply:
I tried it and ran `npx tsc --noEmit` which runs TypeScript type-checker without emitting and build artifacts and it was fine.
```ts
    me: async (_, _params, ctx) => {
      if (!ctx.session.user || !ctx.session.user.username) return null;
      return await getMe(ctx);
    },
```

Reply:
That being said, I don't think session.user.username` can ever be undefined if `user` is defined.